examples/gitserver/README.md

```diff
@@ -1,16 +1,220 @@
+    current project. To see the hostname that was assigned to your Git Server, execute:
+
+    ```sh
+    $ oc get route gitserver
+    ```
+
+    The hostname should be something like: `gitserver-myproject.infra.openshift.com`
+    where `myproject` is the name of your project, and `.infra.openshift.com` is the default
+    suffix for routes in your cluster. In the instructions that follow please substitute
+    `gitserver-myproject.infra.openshift.com` with the host name given to you by your 
+    cluster.
+
+4. Update the PUBLIC_URL of your Git Server with the one provided by the route
+
+    ```sh
+    $ oc env dc/gitserver PUBLIC_URL=http://gitserver-myproject.infra.openshift.com
+    ```
+
+    This will result in a new deployment of the Git Server with the updated environment 
+    variable.
+
+
+### Push code to the Git Server
+
+Code repositories can be initialized on the Git Server by either doing a 'git push' or
+a 'git clone' of the repository.
+
+1. Setup your credentials
+
+   By default the Git Server will allow users or service accounts that can create pods in
+   the project namespace to create and push code to the Git Server. The easiest way to 
+   provide credentials to the Git Server is by using a custom credential helper that will 
+   send your OpenShift token by default to the server.
+
+   ```sh
+   $ git config --global credential.http://gitserver-myproject.infra.openshift.com.helper \
+         '!f() { echo "username=$(oc whoami)"; echo "password=$(oc whoami -t)"; }; f'
+   ```
+
+   **NOTE:** the config key is `credential.[your server URL here].helper`
+
+2. Clone a new repository
+   
+   ```
+   $ git clone http://gitserver-myproject.infra.openshift.com/ruby-hello-world.git
+   ```
+
+   **NOTE:** the ruby-hello-world.git repository does not exist before running this command. 
+   By cloning it, you are creating it in the Git Server.
+
+3. Populate the repository with code
+
+   Add a public Github repository as a remote, fetch it, and push its code to your
+   local Git Server.
+
+   ```sh
+   $ cd ruby-helllo-world
+   $ git remote add github https://github.com/openshift/ruby-hello-world.git
+   $ git fetch github
+   $ git rebase github/master
+   $ git push origin master
+   ```
+
+### Build and Deploy code from the Git Server
+
+1. Create artifacts using `oc new-app`
+
+   ```sh
+   $ oc new-app http://gitserver-myproject.infra.openshift.com/ruby-hello-world.git
+   ```
+
+2. Setup a source secret for the BuildConfig
+
+   In order for an Openshift build to be able to clone code from the gitserver, it must be
+   able to provide credentials to it. The secret for the gitserver service account can be
+   used for this purpose.
+
+   To get the name of a secret, run:
+   ```sh
+   $ oc get secrets | grep gitserver | grep token | head -n 1
+   ```
+
+   Edit the BuildConfig and set the source secret using the secret name returned above.
+   ```sh
+   $ oc edit bc ruby-hello-world
```

Comment:
`oc set secret` with subcommands - we will want to set secrets in a bunch of places